app/org/sagebionetworks/bridge/dynamodb/DynamoInitializer.java

```diff
@@ -216,18 +225,111 @@ static void deleteTable(Class<?> table) {
             final TableDescription table = (new TableDescription())
                     .withTableName(tableName)
                     .withKeySchema(keySchema)
-                    .withAttributeDefinitions(attributes)
+                    .withAttributeDefinitions(attributes.values())
                     .withGlobalSecondaryIndexes(globalIndices)
                     .withLocalSecondaryIndexes(localIndices)
                     .withProvisionedThroughput((new ProvisionedThroughputDescription())
-                            .withReadCapacityUnits(readCapacity)
-                            .withWriteCapacityUnits(writeCapacity));
-            tables.add(table);
+                    .withReadCapacityUnits(readCapacity)
+                    .withWriteCapacityUnits(writeCapacity));
+            tables.put(tableName, table);
         }
-        return tables;
+        return new ArrayList<>(tables.values());
     }
 
     /**
+     * If the method is also annotated with a projection annotation, use the projection it
+     * indicates. For hash/range global indices, this annotation needs to be on the same method
+     * as @DynamoDBIndexHashKey. This is a Bridge-specific annotation, it's not in the AWS SDK. 
+     * @param method
+     * @param indexName
+     * @param descr
+     */
+    private static void addProjectionIfAnnotated(Method method, String indexName, GlobalSecondaryIndexDescription descr) {
+        DynamoDBProjection projection = method.getAnnotation(DynamoDBProjection.class);
+        if (projection != null && indexName.equals(projection.globalSecondaryIndexName())) {
+            descr.setProjection(new Projection().withProjectionType(projection.projectionType()));
+        }
+    }
+
+    private static GlobalSecondaryIndexDescription createGlobalIndexDescr(String indexName, String hashAttrName,
+                    String rangeAttrName) {
+        Preconditions.checkArgument(isNotBlank(indexName));
+        
+        List<KeySchemaElement> keys = Lists.newArrayList();
+        if (hashAttrName != null) {
+            keys.add(new KeySchemaElement(hashAttrName, KeyType.HASH));
+        }
+        if (rangeAttrName != null) {
+            keys.add(new KeySchemaElement(rangeAttrName, KeyType.RANGE));
+        }
+        GlobalSecondaryIndexDescription globalIndex = new GlobalSecondaryIndexDescription()
+            .withIndexName(indexName)
+            .withKeySchema(keys)
+            .withProjection(new Projection().withProjectionType(ProjectionType.KEYS_ONLY))
+            .withProvisionedThroughput(new ProvisionedThroughputDescription()
+            .withReadCapacityUnits(DEFAULT_READ_CAPACITY)
```

Comment:
Use the throughput values specified for the table as a whole, don't just use the defaults.